internal/servers/worker/internal/metric/controller_dialer.go

```diff
@@ -0,0 +1,156 @@
+package metric
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/boundary/globals"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+)
+
+const (
+	labelGrpcCode               = "grpc_code"
+	labelGrpcService            = "grpc_service"
+	labelGrpcMethod             = "grpc_method"
+	workerToControllerSubsystem = "worker_to_controller_cluster"
```

Comment:
IMO the subsystem metric being used is the cluster client (or server service client).  It doesn't really matter who is using the client for the purposes of these metrics.  For example, if we invented another boundary server "proxy" which could handle the requests to this cluster client, it would be nice to not have to rewrite any portion of this code to automatically get the metrics for it.

The name controller is also a bit misleading since there are 2 apis (maybe 3, if you seperate server management service and the session service) exposed by the boundary controller.  These metrics are only for the server management and session services (I think what we've often called the cluster).

Please update here and elsewhere in this file anywhere that refers to the worker and/or calls the the cluster (or server service) client the controller.

Reply:
Addressed [072e97e](https://github.com/hashicorp/boundary/pull/1998/commits/072e97e4eeddfe56f09d75d23e49760605c3b21a)